Gruntfile.js

```diff
@@ -291,12 +303,31 @@ export default Vex;`;
       'webpack:buildProdBravuraOnly',
       'webpack:buildProdGonvilleOnly',
       'webpack:buildProdPetalumaOnly',
-      'copy:moduleJSFiles',
+      'buildESM',
       'copy:modulePackageJSON',
+      'buildTypeDeclarations',
       'typedoc',
     ]
   );
 
+  grunt.registerTask('buildTypeDeclarations', 'Use tsc to create *.d.ts files in build/types/', () => {
+    grunt.log.writeln('Building *.d.ts files in build/types/');
+    const results = child_process.execSync('tsc -p tsconfig.types.json').toString();
+    grunt.log.writeln(results);
+  });
+
+  // Outputs ESM module files to build/esm/.
+  // Also fixes the imports and exports so that they all end in .js.
+  grunt.registerTask('buildESM', 'Use tsc to create ESM JS files in build/esm/', () => {
+    grunt.log.writeln('ESM: Building to build/esm/');
+    const outputTSC = child_process.execSync('tsc -p tsconfig.esm.json').toString();
+    grunt.log.writeln(outputTSC);
+
+    grunt.log.writeln('ESM: Fixing Imports/Exports');
+    const outputFix = child_process.execSync('node ./tools/esm/fix-imports-and-exports.js ./build/esm/').toString();
```

Comment:
After tsc does its job, the resulting JS files still won't work properly as ES modules. That's because our imports don't end with the .js extension. We could actually manually edit all our imports to add js extensions (apparently it's legal in TypeScript, as long as you find the correct webpack plugin to resolve the imports).

Instead, I wrote a script that adds the .js extensions after the fact. This script operates on the `build/esm/**/*.js files` and modifies all imports and exports. It's RegEx. Not the most robust solution, but it works for now.